src/release.rs

```diff
@@ -62,26 +75,28 @@ impl BuildConfig {
         }
     }
 
-    async fn setup_tar_deploy(&self, release_channel: &String) -> Result<PathBuf, Box<dyn Error>> {
-        let file_name = "solana-release";
-        let tar_filename = format!("{file_name}.tar.bz2");
+    async fn setup_tar_deploy(&self, release_channel: &String) -> Result<(), Box<dyn Error>> {
+        let tar_filename = format!("{SOLANA_RELEASE}.tar.bz2");
         self.download_release_from_channel(&tar_filename, release_channel)
             .await?;
 
         // Extract it and load the release version metadata
-        let tarball_filename = self.solana_root_path.join(&tar_filename);
-        let release_dir = self.solana_root_path.join(file_name);
-        extract_release_archive(&tarball_filename, &self.solana_root_path).map_err(|err| {
+        let tarball_filename = self.cluster_root_path.join(&tar_filename);
+        extract_release_archive(&tarball_filename, &self.cluster_root_path).map_err(|err| {
             format!(
                 "Unable to extract {tar_filename} into {}: {err}",
-                release_dir.display()
+                self.install_directory.display()
             )
         })?;
-
-        Ok(release_dir)
+        Ok(())
     }
 
     fn build(&self) -> Result<String, Box<dyn Error>> {
+        let agave_path = match &self.agave_repo_path {
+            Some(path) => path.clone(),
+            None => return Err("Can't build() with agave_repo_path == None".into()),
```

Comment:
micro-nit: maybe make this a bit more human-readable:
```suggestion
            None => return Err("An agave repo path must be configured to build, please specify ... ".into()),
```

Please fill in the correct CLI flag to set